blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/NodeDownstreamBuildAction.java

```diff
@@ -0,0 +1,60 @@
+package io.jenkins.blueocean.service.embedded.rest;
+
+import hudson.model.InvisibleAction;
+import io.jenkins.blueocean.rest.Reachable;
+import io.jenkins.blueocean.rest.hal.Link;
+import org.jenkinsci.plugins.workflow.actions.FlowNodeAction;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
+
+import java.util.Objects;
+
+/**
+ * Annotates a FlowNode to point to a downstream build triggered by said node. Applied by
+ * io.jenkins.blueocean.listeners.DownstreamJobListener in blueocean-pipeline-api-impl
+ */
+@ExportedBean
+public class NodeDownstreamBuildAction extends InvisibleAction implements FlowNodeAction, Reachable, BluePipelineAction {
```

Comment:
Any reason we need it in blueocean-rest-impl? It should be moved to blueocean-pipeline-api-impl as thats only place its needed. All Flow node specific code exists only in blueocean-pipeline-api-impl.

Reply:
There *was* a reason, but I can't remember it and it may no longer be valid so I'll move it and see if there's any sort of earth-shattering kaboom :)